### PR TITLE
Cmdline output filter

### PR DESCRIPTION
**Pull request by olegklimov**

### CODE REVIEW

File: src/tools/tool_cmdline.rs [hunk 1]
```diff
@@ -38,6 +38,7 @@ struct CmdlineToolConfig {
     #[serde(default = "_default_timeout")]
     timeout: u64,
     background: Option<CmdlineToolBackground>,
+    output_filter: Option<crate::postprocessing::pp_command_output::CmdlineOutputFilter>,
```
Comment (valaises) [inline src/tools/tool_cmdline.rs:41]: import it in the top as crate::postprocessing::pp_command_output::CmdlineOutputFilter occurs 3 !!! times in the file

File: src/postprocessing/pp_command_output.rs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+fn default_limit_chars() -> usize {
+    10000
+}
+
+fn default_valuable_top_or_bottom() -> String {
+    "top".to_string()
+}
+
+fn default_grep() -> String {
```
Comment (valaises) [inline src/postprocessing/pp_command_output.rs:46]: why default grep is so weird?

File: src/postprocessing/pp_command_output.rs [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    pub limit_chars: usize,
+    #[serde(default = "default_valuable_top_or_bottom")]
+    pub valuable_top_or_bottom: String,
+    #[serde(default = "default_grep")]
+    pub grep: String,
+    #[serde(default = "default_grep_context_lines")]
+    pub grep_context_lines: usize,
+    #[serde(default = "default_remove_from_output")]
+    pub remove_from_output: String,
+}
```
Comment (valaises) [inline src/postprocessing/pp_command_output.rs:19]: I have no clue what it does and I will have no idea how to use it
what is remove_from_output?
what grep_context_lines is?
what if I don't want grep

File: src/integrations/mod.rs [hunk 4]
```diff
@@ -50,19 +50,29 @@ chrome:
...
-//     description: "Run `make` inside a C/C++ project, or a similar project with a Makefile."
-//     parameters:
-//       - name: "project_path"
-//         description: "absolute path to the project"
+
+# Command line: things you can call and immediately get an answer
+#cmdline:
+#  run_make:
+#    command: "make"
+#    command_workdir: "%project_path%"
```
Comment (valaises) [inline src/integrations/mod.rs:64]: why didn't you leave a template structure "make %project_path%"?

**Review by valaises — CHANGES_REQUESTED** [review]

File: src/integrations/mod.rs (hunk 4 above)
Comment (olegklimov) [reply src/integrations/mod.rs:64]: because this is how you call `make`
try to recall -C without googling

File: src/tools/tool_cmdline.rs (hunk 1 above)
Comment (olegklimov) [reply src/tools/tool_cmdline.rs:41]: I like to see the full path

Comment (valaises) [reply src/tools/tool_cmdline.rs:41]: I dislike when this string of 40 chars repeats 3 bloody times in a code

File: src/integrations/mod.rs (hunk 4 above)
Comment (valaises) [reply src/integrations/mod.rs:64]: this structure is not abstract enough that I would be able to execute any command

**Pull request closed without merging**